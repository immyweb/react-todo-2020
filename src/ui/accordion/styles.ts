import glamorous from 'glamorous';
import { colours } from '../../styles/colours';
import { mediaQueries } from '../../styles/breakpoints';
import { transparentize } from 'polished';

import { H4, H5, A } from '../../ui/typography/typography';

import { ITheme } from '../../types/theme';

// Accordion

export const AccordionContainer = glamorous.section({
  label: 'accordion-container',
  position: 'relative',
  marginBottom: 25,
  background: colours.$white,
  ':after': {
    clear: 'both',
    content: '""',
    display: 'table',
  },
});

export const Border = glamorous.div<{ colour: string }>(
  {
    label: 'border',
    left: 0,
    content: '""',
    position: 'absolute',
    top: 0,
    height: '100%',
    width: 5,
    zIndex: 10,
  },
  ({ colour }) => ({
    background: colour,
  }),
);

// Accordion Item

export const Item = glamorous.div<{
  open: boolean;
  index: number;
  theme: ITheme;
  border?: boolean;
}>(
  {
    label: 'accordion-item',
    position: 'relative',
    overflowY: 'hidden',
    transition: 'border-color 250ms ease',
    ':before': {
      content: '""',
      position: 'absolute',
      height: 0,
      width: 0,
      top: 22,
      left: 5,
      border: '4px solid transparent',
    },
    ':last-child': {
      borderBottom: 0,
    },
  },
  ({ open, index, theme, border }) => ({
    borderLeft: border ? `5px solid ${theme.secondary}` : 'none',
    borderBottom: open
      ? `2px solid ${transparentize(0.8, theme.primary)}`
      : `2px solid ${colours.$white}`,
    label: `${index}--accordion-item`,
  }),
);

// Accordion Item Head

export const Header = glamorous.header<{ open: boolean; colour: string }>(
  {
    label: 'accordion-header',
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    padding: '15px 20px',
    minHeight: 80,
    color: colours.$offBlack,
    ':hover': {
      cursor: 'pointer',
    },
  },
  ({ open, colour }) => ({
    background: open ? colours.$white : transparentize(0.8, colour),
  }),
);

export const BadgeHolder = glamorous.div({
  label: 'badge-holder',
  display: 'flex',
  borderRadius: '50%',
  overflow: 'hidden',
  position: 'relative',
});

export const Badge = glamorous.img({
  label: 'badge',
  width: 50,
  height: 50,
});

export const TextHolder = glamorous.div<{ hasBadge?: boolean }>(
  {
    label: 'title-text',
    flexDirection: 'column',
    position: 'relative',
    paddingRight: 15,
  },
  ({ hasBadge }) => ({
    paddingLeft: hasBadge ? 15 : 0,
    flexBasis: hasBadge ? `calc(100% - 65px)` : '100%',
  }),
);

export const SubTitle = glamorous(H5)({
  label: 'accordion-date',
  marginBottom: 0,
});

export const Title = glamorous(H4)<{ index: number }>(
  {
    label: 'heading',
    marginBottom: 0,
    position: 'relative',
  },
  ({ index }) => ({
    label: `${index}--accordion-h4`,
  }),
);

// Accordion Item Body

export const defaultStyle = {
  transition: `opacity ${150}ms ease-out`,
  opacity: 0,
};

export const transitionStyles = {
  entering: { opacity: 0 },
  entered: { opacity: 1 },
  exiting: { opacity: 1 },
  exited: { opacity: 0 },
};

export const Panel = glamorous.article<{ open: boolean }>(
  {
    label: 'accordion-panel',
    padding: '0 20px',
    overflowY: 'hidden',
    [mediaQueries.tabletPortrait]: {
      borderBottom: 0,
    },
  },
  ({ open }) => ({
    display: open ? 'block' : 'none',
  }),
);

export const Copy = glamorous.div<{ colour: string }>(
  {
    label: 'accordion-copy',
    margin: '0 0 15px',
    '& a': {
      textDecoration: 'none',
    },
  },
  ({ colour }) => ({
    '& a': {
      color: colour,
    },
  }),
);

export const Link = glamorous(A)<{ colour: string }>(
  {
    label: 'accordion-link',
    display: 'inline-block',
  },
  ({ colour }) => ({
    color: colour,
  }),
);
