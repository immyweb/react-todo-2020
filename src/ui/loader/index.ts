import { Loader } from '../../ui/loader/loader';

export default Loader;
