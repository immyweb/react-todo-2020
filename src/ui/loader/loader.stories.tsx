// import * as React from 'react';
// import { storiesOf } from '@storybook/react';
// import { colours } from '../../styles/colours';

// import { Loader } from './loader';

// const stories = storiesOf('Loader', module);

// stories.add('default useage', () => (
//   <div style={{ height: '100vh' }}>
//     <Loader />
//   </div>
// ));

// stories.add('custom background colour', () => (
//   <div style={{ height: '100vh' }}>
//     <Loader backgroundColour={colours.$fabulous} />
//   </div>
// ));

// stories.add('custom dot colour', () => (
//   <div style={{ height: '100vh' }}>
//     <Loader dotColour={colours.$fabulous} />
//   </div>
// ));
