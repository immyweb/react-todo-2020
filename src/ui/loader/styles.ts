import glamorous from 'glamorous';
import { keyframes } from 'glamor';

interface IGProps {
  backgroundColour?: string;
  dotColour?: string;
}

export const LoaderWrapper = glamorous.div<IGProps>(
  {
    label: 'loader-wrapper',
    position: 'relative',
    width: '100%',
    height: '100%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    minHeight: 200,
  },
  ({ backgroundColour }) => ({
    backgroundColor: backgroundColour,
  }),
);

const slideKeyframes = keyframes({
  '0%': {
    opacity: 0,
    transform: `translateX(-300px)`,
  },
  '33%': {
    opacity: 1,
    transform: `translateX(0px)`,
  },
  '66%': {
    opacity: 1,
    transform: `translateX(0px)`,
  },
  '100%': {
    opacity: 0,
    transform: `translateX(300px)`,
  },
});

const slideTime = '3s';
const slideSettings = 'infinite ease-in-out';

export const Dot = glamorous.span<IGProps>(
  {
    label: 'dot',
    fontSize: 20,
    opacity: 0,
    '&:nth-child(6)': {
      animation: `${slideKeyframes} ${slideTime} ${slideSettings}`,
    },
    '&:nth-child(5)': {
      animation: `${slideKeyframes} ${slideTime} 100ms ${slideSettings}`,
    },
    '&:nth-child(4)': {
      animation: `${slideKeyframes} ${slideTime} 200ms ${slideSettings}`,
    },
    '&:nth-child(3)': {
      animation: `${slideKeyframes} ${slideTime} 300ms ${slideSettings}`,
    },
    '&:nth-child(2)': {
      animation: `${slideKeyframes} ${slideTime} 400ms ${slideSettings}`,
    },
    '&:nth-child(1)': {
      animation: `${slideKeyframes} ${slideTime} 500ms ${slideSettings}`,
    },
  },
  ({ dotColour }) => ({
    color: dotColour,
  }),
);
