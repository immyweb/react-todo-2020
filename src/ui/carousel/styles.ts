import glamorous from 'glamorous';

import { colours } from '../../styles/colours';
import { fonts } from '../../styles/fonts';
import { BtnNoStyle } from '../../styles/globals';
import { Direction } from './types';

export const CarouselContainer = glamorous.section({
  label: 'carousel-container',
  paddingBottom: 33,
  marginBottom: 25,
  position: 'relative',
  width: '100%',
});

export const Slide = glamorous.div({
  label: 'carousel-slide',
  cursor: 'pointer',
  padding: '0 10px',
  ':focus': {
    outline: 'none',
  },
});

export const SlideLink = glamorous.a({
  label: 'carousel-link',
  display: 'inline-block',
  width: '100%',
  height: '100%',
  objectFit: 'contain',
  textDecoration: 'none',
});

export const SlideImg = glamorous.img<{ borderColour: string }>(
  {
    label: 'carousel-image',
    width: '100%',
    height: '100%',
    objectFit: 'contain',
  },
  ({ borderColour }) => ({
    borderBottom: `3px solid ${borderColour}`,
  }),
);

export const SlideLabel = glamorous.p<{ txtColour: string }>(
  {
    label: 'carousel-label',
    textAlign: 'center',
    fontFamily: fonts.$theSunHeavyNarrow,
    fontSize: 18,
    paddingTop: 8,
  },
  ({ txtColour }) => ({
    color: txtColour,
  }),
);

export const SlideNav = glamorous(BtnNoStyle)<{ direction: Direction }>(
  {
    label: 'carousel-nav',
    position: 'absolute',
    zIndex: 3,
    bottom: -47,
    width: 48,
    height: 48,
    ':hover': {
      cursor: 'pointer',
    },
  },
  ({ direction }) => ({
    left: direction === Direction.PREV ? 25 : 'auto',
    right: direction === Direction.NEXT ? 25 : 'auto',
  }),
);

// Pagination

export const PaginationContainer = glamorous.div({
  label: 'pagination-container',
  position: 'absolute',
  bottom: 0,
  zIndex: 2,
  width: '100%',
  height: 20,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
});

export const Dot = glamorous.span<{ colour?: string; label?: string }>(
  {
    display: 'inline-block',
    borderRadius: '50%',
    margin: '0 5px',
  },
  ({ colour, label }) => ({
    label,
    background: colour ? colour : colours.$news,
  }),
);
