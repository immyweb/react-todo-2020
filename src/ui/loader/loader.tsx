import * as React from 'react';

import { colours } from '../../styles/colours';

import { LoaderWrapper, Dot } from './styles';

interface IProps {
  backgroundColour?: string;
  dotColour?: string;
}

export const Loader: React.SFC<IProps> = ({ backgroundColour, dotColour }) => {
  return (
    <LoaderWrapper backgroundColour={backgroundColour ? backgroundColour : 'none'}>
      <Dot dotColour={dotColour ? dotColour : colours.$lightGrey}>●</Dot>
      <Dot dotColour={dotColour ? dotColour : colours.$lightGrey}>●</Dot>
      <Dot dotColour={dotColour ? dotColour : colours.$lightGrey}>●</Dot>
      <Dot dotColour={dotColour ? dotColour : colours.$lightGrey}>●</Dot>
      <Dot dotColour={dotColour ? dotColour : colours.$lightGrey}>●</Dot>
      <Dot dotColour={dotColour ? dotColour : colours.$lightGrey}>●</Dot>
    </LoaderWrapper>
  );
};
