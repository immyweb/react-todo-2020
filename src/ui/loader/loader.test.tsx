import * as React from 'react';
import { shallow } from 'enzyme';
import { Loader } from './loader';

import { colours } from '../../styles/colours';

import { LoaderWrapper, Dot } from './styles';

describe('<Loader />', () => {
  it('renders ok', () => {
    const wrapper = shallow(<Loader />);
    const loaderWrapper = wrapper.find(LoaderWrapper);
    expect(loaderWrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('has 6 dots', () => {
    const wrapper = shallow(<Loader />);
    const dot = wrapper.find(Dot);
    expect(dot.length).toBe(6);
    expect(wrapper).toMatchSnapshot();
  });

  it('accepts a background colour prop', () => {
    const wrapper = shallow(<Loader backgroundColour={colours.$fabulous} />);
    const backgroundColour = wrapper.find(LoaderWrapper).props().backgroundColour;
    expect(backgroundColour).toEqual('#e665bf');
    expect(wrapper).toMatchSnapshot();
  });

  it('accepts a dot colour prop', () => {
    const wrapper = shallow(<Loader dotColour={colours.$fabulous} />);

    const dotColour = wrapper
      .find(Dot)
      .at(1)
      .props().dotColour;
    expect(dotColour).toEqual('#e665bf');
    expect(wrapper).toMatchSnapshot();
  });
});
