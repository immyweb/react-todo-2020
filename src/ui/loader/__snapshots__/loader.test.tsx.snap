// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> accepts a background colour prop 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    backgroundColour="#e665bf"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "backgroundColour": "#e665bf",
      "children": Array [
        <glamorous(span)
          dotColour="#e6eaeb"
        >
          ●
        </glamorous(span)>,
        <glamorous(span)
          dotColour="#e6eaeb"
        >
          ●
        </glamorous(span)>,
        <glamorous(span)
          dotColour="#e6eaeb"
        >
          ●
        </glamorous(span)>,
        <glamorous(span)
          dotColour="#e6eaeb"
        >
          ●
        </glamorous(span)>,
        <glamorous(span)
          dotColour="#e6eaeb"
        >
          ●
        </glamorous(span)>,
        <glamorous(span)
          dotColour="#e6eaeb"
        >
          ●
        </glamorous(span)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "●",
          "dotColour": "#e6eaeb",
        },
        "ref": null,
        "rendered": "●",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "●",
          "dotColour": "#e6eaeb",
        },
        "ref": null,
        "rendered": "●",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "●",
          "dotColour": "#e6eaeb",
        },
        "ref": null,
        "rendered": "●",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "●",
          "dotColour": "#e6eaeb",
        },
        "ref": null,
        "rendered": "●",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "●",
          "dotColour": "#e6eaeb",
        },
        "ref": null,
        "rendered": "●",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "●",
          "dotColour": "#e6eaeb",
        },
        "ref": null,
        "rendered": "●",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "backgroundColour": "#e665bf",
        "children": Array [
          <glamorous(span)
            dotColour="#e6eaeb"
          >
            ●
          </glamorous(span)>,
          <glamorous(span)
            dotColour="#e6eaeb"
          >
            ●
          </glamorous(span)>,
          <glamorous(span)
            dotColour="#e6eaeb"
          >
            ●
          </glamorous(span)>,
          <glamorous(span)
            dotColour="#e6eaeb"
          >
            ●
          </glamorous(span)>,
          <glamorous(span)
            dotColour="#e6eaeb"
          >
            ●
          </glamorous(span)>,
          <glamorous(span)
            dotColour="#e6eaeb"
          >
            ●
          </glamorous(span)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "●",
            "dotColour": "#e6eaeb",
          },
          "ref": null,
          "rendered": "●",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "●",
            "dotColour": "#e6eaeb",
          },
          "ref": null,
          "rendered": "●",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "●",
            "dotColour": "#e6eaeb",
          },
          "ref": null,
          "rendered": "●",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "●",
            "dotColour": "#e6eaeb",
          },
          "ref": null,
          "rendered": "●",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "●",
            "dotColour": "#e6eaeb",
          },
          "ref": null,
          "rendered": "●",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "●",
            "dotColour": "#e6eaeb",
          },
          "ref": null,
          "rendered": "●",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Loader /> accepts a dot colour prop 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    dotColour="#e665bf"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "backgroundColour": "none",
      "children": Array [
        <glamorous(span)
          dotColour="#e665bf"
        >
          ●
        </glamorous(span)>,
        <glamorous(span)
          dotColour="#e665bf"
        >
          ●
        </glamorous(span)>,
        <glamorous(span)
          dotColour="#e665bf"
        >
          ●
        </glamorous(span)>,
        <glamorous(span)
          dotColour="#e665bf"
        >
          ●
        </glamorous(span)>,
        <glamorous(span)
          dotColour="#e665bf"
        >
          ●
        </glamorous(span)>,
        <glamorous(span)
          dotColour="#e665bf"
        >
          ●
        </glamorous(span)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "●",
          "dotColour": "#e665bf",
        },
        "ref": null,
        "rendered": "●",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "●",
          "dotColour": "#e665bf",
        },
        "ref": null,
        "rendered": "●",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "●",
          "dotColour": "#e665bf",
        },
        "ref": null,
        "rendered": "●",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "●",
          "dotColour": "#e665bf",
        },
        "ref": null,
        "rendered": "●",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "●",
          "dotColour": "#e665bf",
        },
        "ref": null,
        "rendered": "●",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "●",
          "dotColour": "#e665bf",
        },
        "ref": null,
        "rendered": "●",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "backgroundColour": "none",
        "children": Array [
          <glamorous(span)
            dotColour="#e665bf"
          >
            ●
          </glamorous(span)>,
          <glamorous(span)
            dotColour="#e665bf"
          >
            ●
          </glamorous(span)>,
          <glamorous(span)
            dotColour="#e665bf"
          >
            ●
          </glamorous(span)>,
          <glamorous(span)
            dotColour="#e665bf"
          >
            ●
          </glamorous(span)>,
          <glamorous(span)
            dotColour="#e665bf"
          >
            ●
          </glamorous(span)>,
          <glamorous(span)
            dotColour="#e665bf"
          >
            ●
          </glamorous(span)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "●",
            "dotColour": "#e665bf",
          },
          "ref": null,
          "rendered": "●",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "●",
            "dotColour": "#e665bf",
          },
          "ref": null,
          "rendered": "●",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "●",
            "dotColour": "#e665bf",
          },
          "ref": null,
          "rendered": "●",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "●",
            "dotColour": "#e665bf",
          },
          "ref": null,
          "rendered": "●",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "●",
            "dotColour": "#e665bf",
          },
          "ref": null,
          "rendered": "●",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "●",
            "dotColour": "#e665bf",
          },
          "ref": null,
          "rendered": "●",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Loader /> has 6 dots 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "backgroundColour": "none",
      "children": Array [
        <glamorous(span)
          dotColour="#e6eaeb"
        >
          ●
        </glamorous(span)>,
        <glamorous(span)
          dotColour="#e6eaeb"
        >
          ●
        </glamorous(span)>,
        <glamorous(span)
          dotColour="#e6eaeb"
        >
          ●
        </glamorous(span)>,
        <glamorous(span)
          dotColour="#e6eaeb"
        >
          ●
        </glamorous(span)>,
        <glamorous(span)
          dotColour="#e6eaeb"
        >
          ●
        </glamorous(span)>,
        <glamorous(span)
          dotColour="#e6eaeb"
        >
          ●
        </glamorous(span)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "●",
          "dotColour": "#e6eaeb",
        },
        "ref": null,
        "rendered": "●",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "●",
          "dotColour": "#e6eaeb",
        },
        "ref": null,
        "rendered": "●",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "●",
          "dotColour": "#e6eaeb",
        },
        "ref": null,
        "rendered": "●",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "●",
          "dotColour": "#e6eaeb",
        },
        "ref": null,
        "rendered": "●",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "●",
          "dotColour": "#e6eaeb",
        },
        "ref": null,
        "rendered": "●",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "●",
          "dotColour": "#e6eaeb",
        },
        "ref": null,
        "rendered": "●",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "backgroundColour": "none",
        "children": Array [
          <glamorous(span)
            dotColour="#e6eaeb"
          >
            ●
          </glamorous(span)>,
          <glamorous(span)
            dotColour="#e6eaeb"
          >
            ●
          </glamorous(span)>,
          <glamorous(span)
            dotColour="#e6eaeb"
          >
            ●
          </glamorous(span)>,
          <glamorous(span)
            dotColour="#e6eaeb"
          >
            ●
          </glamorous(span)>,
          <glamorous(span)
            dotColour="#e6eaeb"
          >
            ●
          </glamorous(span)>,
          <glamorous(span)
            dotColour="#e6eaeb"
          >
            ●
          </glamorous(span)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "●",
            "dotColour": "#e6eaeb",
          },
          "ref": null,
          "rendered": "●",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "●",
            "dotColour": "#e6eaeb",
          },
          "ref": null,
          "rendered": "●",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "●",
            "dotColour": "#e6eaeb",
          },
          "ref": null,
          "rendered": "●",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "●",
            "dotColour": "#e6eaeb",
          },
          "ref": null,
          "rendered": "●",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "●",
            "dotColour": "#e6eaeb",
          },
          "ref": null,
          "rendered": "●",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "●",
            "dotColour": "#e6eaeb",
          },
          "ref": null,
          "rendered": "●",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Loader /> renders ok 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "backgroundColour": "none",
      "children": Array [
        <glamorous(span)
          dotColour="#e6eaeb"
        >
          ●
        </glamorous(span)>,
        <glamorous(span)
          dotColour="#e6eaeb"
        >
          ●
        </glamorous(span)>,
        <glamorous(span)
          dotColour="#e6eaeb"
        >
          ●
        </glamorous(span)>,
        <glamorous(span)
          dotColour="#e6eaeb"
        >
          ●
        </glamorous(span)>,
        <glamorous(span)
          dotColour="#e6eaeb"
        >
          ●
        </glamorous(span)>,
        <glamorous(span)
          dotColour="#e6eaeb"
        >
          ●
        </glamorous(span)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "●",
          "dotColour": "#e6eaeb",
        },
        "ref": null,
        "rendered": "●",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "●",
          "dotColour": "#e6eaeb",
        },
        "ref": null,
        "rendered": "●",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "●",
          "dotColour": "#e6eaeb",
        },
        "ref": null,
        "rendered": "●",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "●",
          "dotColour": "#e6eaeb",
        },
        "ref": null,
        "rendered": "●",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "●",
          "dotColour": "#e6eaeb",
        },
        "ref": null,
        "rendered": "●",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "●",
          "dotColour": "#e6eaeb",
        },
        "ref": null,
        "rendered": "●",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "backgroundColour": "none",
        "children": Array [
          <glamorous(span)
            dotColour="#e6eaeb"
          >
            ●
          </glamorous(span)>,
          <glamorous(span)
            dotColour="#e6eaeb"
          >
            ●
          </glamorous(span)>,
          <glamorous(span)
            dotColour="#e6eaeb"
          >
            ●
          </glamorous(span)>,
          <glamorous(span)
            dotColour="#e6eaeb"
          >
            ●
          </glamorous(span)>,
          <glamorous(span)
            dotColour="#e6eaeb"
          >
            ●
          </glamorous(span)>,
          <glamorous(span)
            dotColour="#e6eaeb"
          >
            ●
          </glamorous(span)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "●",
            "dotColour": "#e6eaeb",
          },
          "ref": null,
          "rendered": "●",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "●",
            "dotColour": "#e6eaeb",
          },
          "ref": null,
          "rendered": "●",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "●",
            "dotColour": "#e6eaeb",
          },
          "ref": null,
          "rendered": "●",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "●",
            "dotColour": "#e6eaeb",
          },
          "ref": null,
          "rendered": "●",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "●",
            "dotColour": "#e6eaeb",
          },
          "ref": null,
          "rendered": "●",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "●",
            "dotColour": "#e6eaeb",
          },
          "ref": null,
          "rendered": "●",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
